Java/core/src/main/java/com/amazon/randomcutforest/PointStoreCoordinator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest;
+
+import static com.amazon.randomcutforest.CommonUtils.checkNotNull;
+
+import java.util.List;
+
+import com.amazon.randomcutforest.store.PointStore;
+
+public class PointStoreCoordinator implements IUpdateCoordinator<Sequential<Integer>> {
```

Comment:
minor. is it a good idea to have a pair of sequential and updateInput so the updateInput doesn't need to be stored as a state, different from sequential?

Reply:
Sorry, I don't understand the suggestion.

Reply:
I think I see your point. In the next revision I've changed the method signature of the `completeUpdate` method to:

```
public abstract void completeUpdate(P updateInput, List<P> deletedPoints);
```

The new implementation of `update` looks like this:

```
public void update(double[] point) {
        double[] pointCopy = cleanCopy(point);
        P updateInput = updateCoordinator.initUpdate(pointCopy);
        List<P> results = update(updateInput);
        updateCoordinator.completeUpdate(updateInput, results);
}
```

So now implementations of IUpdateCoordinator don't need to keep an instance variable to store any part of the updateInput value.